### PR TITLE
Fix lost immutable timestamp lock, v2

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**:
- see #4449, this is the same PR but not on a fork which buildsystem doesn't like
- added tests to changes in 4449

**Implementation Description (bullets)**:
- Port changes from #4449 
- Add simple tests

**Testing (What was existing testing like?  What have you done to improve it?)**:
- Added 2 new tests

**Concerns (what feedback would you like?)**:
- Did I accidentally make a typo or copy something incorrectly?

**Where should we start reviewing?**: LegacyTimelockService

**Priority (whenever / two weeks / yesterday)**: see #4449

### CODE REVIEW

File: lock-impl/src/test/java/com/palantir/lock/impl/LegacyTimelockServiceTest.java [hunk 1]
```diff
@@ -133,6 +136,41 @@ public void lockDelegatesToLockService() throws InterruptedException {
...
 
+    @Test
+    public void unlocksImmutableTimestampLockIfGettingMinLockedInVersionThrows() throws InterruptedException {
+        Exception illegalStateException = new IllegalStateException();
+        when(timestampService.getFreshTimestamp()).thenReturn(5L);
+        when(lockService.lock(eq(LOCK_CLIENT.getClientId()), any())).thenReturn(LOCK_REFRESH_TOKEN);
+        when(lockService.getMinLockedInVersionId(any(LockClient.class))).thenThrow(illegalStateException);
+
+        try {
+            timelock.lockImmutableTimestamp();
```
Comment (carterkozak) [inline lock-impl/src/test/java/com/palantir/lock/impl/LegacyTimelockServiceTest.java]: Nit: assertThatThrownBy

File: lock-impl/src/test/java/com/palantir/lock/impl/LegacyTimelockServiceTest.java [hunk 2]
```diff
@@ -133,6 +136,41 @@ public void lockDelegatesToLockService() throws InterruptedException {
...
+        Exception illegalStateException = new IllegalStateException();
+        when(timestampService.getFreshTimestamp())
+                .thenReturn(5L) // needed because locking the immutable timestamp takes out a fresh ts
+                .thenThrow(illegalStateException);
+        when(lockService.lock(eq(LOCK_CLIENT.getClientId()), any())).thenReturn(LOCK_REFRESH_TOKEN);
+
+        try {
+            timelock.startIdentifiedAtlasDbTransaction();
+            fail();
+        } catch (Exception e) {
```
Comment (carterkozak) [inline lock-impl/src/test/java/com/palantir/lock/impl/LegacyTimelockServiceTest.java]: Here as well

**Comment by GrahamDennis** [comment]
Happy new year @jeremyk-91! (ping)

**Comment by jeremyk-91** [comment]
Hey @GrahamDennis, happy new year! This should be in very shortly

**Comment by carterkozak** [comment]
thanks 👍

**Pull request merged**